libmuscle/cpp/src/libmuscle/instance.cpp

```diff
@@ -923,7 +926,40 @@ Instance::Instance(
 #endif
         )
     : pimpl_(new Impl(
-                argc, argv, ports
+                argc, argv, ports, InstanceFlags::NONE
+#ifdef MUSCLE_ENABLE_MPI
+                , communicator, root
+#endif
+                ))
+{}
+
+Instance::Instance(
+        int argc, char const * const argv[],
+        InstanceFlags flags
+#ifdef MUSCLE_ENABLE_MPI
```

Comment:
We could potentially make two macros like this and save some repetition, but I'm not sure it would actually make things more readable, what do you think?

```
#ifdef MUSCLE_ENABLE_MPI
#define MPI_ARGS_DECL , MPI_Comm const & communicator, int root
#define MPI_ARGS_CALL , communicator, root
#else
#define MPI_ARGS_DECL
#define MPI_ARGS_CALL
#endif
```
and then have
```
Instance::Instance(
        int argc, char const * const argv[],
        PortsDescription const & ports,
        InstanceFlags flags
        MPI_ARGS_DECL
        )
    : pimpl_(new Impl(
                argc, argv, ports, flags MPI_ARGS_CALL))
{}
```
```

Reply:
I've applied it in `instance.cpp` and now all Instance constructor implementations fit on a single screen (including the macro definitions).

Although it makes the calls harder to read, I still think it's an improvement compared to all `#ifdef`s that were previously littering the place 😄